cmd/egress-v4-cni-plugin/cni.go

```diff
@@ -0,0 +1,433 @@
+	if err != nil {
+		log.Debugf("failed to setup container Veth: %v", err)
+		return err
+	}
+
+	if err = setupHostVeth(hostInterface.Name, tmpResult); err != nil {
+		return err
+	}
+
+	log.Debugf("Node IP: %s", netConf.NodeIP)
+	if netConf.NodeIP != nil {
+		for _, ipc := range tmpResult.IPs {
+			if ipc.Version == "4" {
+				//log.Printf("Configuring SNAT %s -> %s", ipc.Address.IP, netConf.SnatIP)
+				if err := snat.Snat4(netConf.NodeIP, ipc.Address.IP, chain, comment); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	//Copy interfaces over to result, but not IPs.
+	result.Interfaces = append(result.Interfaces, tmpResult.Interfaces...)
+	//Note: Useful for debug, will do away with the below log prior to release
+	for _,v := range result.IPs {
+		log.Debugf("Interface Name: %v; IP: %s", v.Interface, v.Address)
+	}
+
+	// Pass through the previous result
+	return types.PrintResult(result, netConf.CNIVersion)
+}
+
+func cmdDel(args *skel.CmdArgs) error {
+	netConf, log, err := loadConf(args.StdinData)
+	if err != nil {
+		return fmt.Errorf("failed to parse config: %v", err)
+	}
+
+	//We only need this plugin to kick in if v6 is enabled
+	if netConf.Enabled == "false" {
+		return nil
+	}
+
+	log.Debugf("Received Del Request: conf=%v", netConf)
+	if err := ipam.ExecDel(netConf.IPAM.Type, args.StdinData); err != nil {
+		log.Debugf("running IPAM plugin failed: %v", err)
+		return fmt.Errorf("running IPAM plugin failed: %v", err)
+	}
+
+	ipnets := []*net.IPNet{}
+	if args.Netns != "" {
+		err := ns.WithNetNSPath(args.Netns, func(hostNS ns.NetNS) error {
+			var err error
+
+			// DelLinkByNameAddr function deletes an interface and returns IPs assigned to it but it
+			// excludes IPs that are not global unicast addresses (or) private IPs. Will not work for
+			// our scenario as we use 169.254.0.0/16 range for v4 IPs.
+
+			//Get the interface we want to delete
+			iface, err := netlink.LinkByName(netConf.IfName)
+
+			if err != nil {
+				if _, ok := err.(netlink.LinkNotFoundError); ok {
+					return nil
+				}
+				return nil
+			}
+
+			//Retrieve IP addresses assigned to the interface
+			addrs, err := netlink.AddrList(iface, netlink.FAMILY_ALL)
+			if err != nil {
+				return fmt.Errorf("failed to get IP addresses for %q: %v", netConf.IfName, err)
+			}
+
+			//Delete the interface/link.
+			if err = netlink.LinkDel(iface); err != nil {
+				return fmt.Errorf("failed to delete %q: %v", netConf.IfName, err)
+			}
+
+			for _, addr := range addrs {
+				ipnets = append(ipnets, addr.IPNet)
+			}
+
+			if err != nil && err == ip.ErrLinkNotFound {
+				log.Debugf("DEL: Link Not Found, returning", err)
+				return nil
+			}
+			return err
+		})
+
+		//DEL should be best effort. We should clean up as much as we can and avoid returning error
+		//CNI Spec: TODO
+		if err != nil {
+			log.Debugf("DEL: Executing in container ns errored out, returning", err)
+			return err
+		}
```

Comment:
As the comment above says, I think it's important that we continue here.   Most errors in this particular section of code will be because the namespace is already gone (or similar), and we don't want to loop forever on this 'not an error' error scenario.

```suggestion
		//DEL should be best effort. We should clean up as much as we can and avoid returning error
		if err != nil {
			log.Debugf("DEL: Executing in container ns errored out, ignoring %v", err)
		}
```